app/controllers/facility_accounts_controller.rb

```diff
@@ -52,7 +52,10 @@ def initialize
 
   # GET /facilties/:facility_id/accounts
   def index
-    @accounts = Account.has_orders_for_facility(current_facility).paginate(:page => params[:page])
+    accounts = Account.with_orders_for_facility(current_facility)
+    accounts = accounts.where(facility_id: nil) if current_facility.cross_facility?
```

Comment:
This is the only place we use `.with_orders_for_facility` (which is why I felt safe changing the name). And since it's used nowhere else I considered making the method do what I'm doing here, to filter out facility-specific accounts when the facility is "all". But would that method name still make sense if it did that? I'm not sure what I'd call it.